test/jdk/java/io/SequenceInputStream/TransferTo.java

```diff
@@ -126,6 +127,40 @@ public void testStreamContents() throws Exception {
                 outputStreamProvider, createRandomBytes(4096, 0), 0, 4096);
     }
 
+    /*
+     * Special case: Assert subsequent input stream is read when preceding stream already was MAX_VALUE long.
+     * Note: Not testing actual content as it requires multiple GBs of memory and long time.
+     */
+    @Test
+    public void testHugeStream() throws Exception {
+        InputStream is1 = repeat(0, Integer.MAX_VALUE);
+        InputStream is2 = repeat(0, 1);
+        assertEquals(is1.available(), Integer.MAX_VALUE);
+        assertEquals(is2.available(), 1);
+        SequenceInputStream sis = new SequenceInputStream(is1, is2);
+        OutputStream nos = OutputStream.nullOutputStream();
+        sis.transferTo(nos);
+        assertEquals(is1.available(), 0);
+        assertEquals(is2.available(), 0);
+    }
+
+    private static InputStream repeat(int b, long count) {
+        return new InputStream() {
+            private int pos;
+            @Override
+            public int available() throws IOException {
+                return (int) Math.min(count - pos, Integer.MAX_VALUE);
+            }
+            @Override
+            public int read() throws IOException {
+                if (pos >= count)
+                    return -1;
+                pos++;
+                return b;
+            }
```

Comment:
To trigger the problematic case, we need to transfer more than `Long.MAX_VALUE`. `Integer.MAX_VALUE` is not enough, and we need nearly `Integer.MAX_VALUE * Integer.MAX_VALUE` bytes to trigger the problematic case which might take noticeable time.

I think the easiest way would be overriding `transferTo(...)` in a mock `InputStream` so it returns `Long.MAX_VALUE`, and adding that stream in a sequence.

Reply:
I know and the solution is already in progress (see my last comment from yesterday). Stay tuned.

Reply:
Done, see https://github.com/openjdk/jdk/pull/17119/commits/556a8dc4d3d8ff157e8374625eb129a08926e0dd. :-)